pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -343,6 +343,11 @@ def _evaluate(
                     m = 'only EvalResults or dicts are allowed from validation_step'
                     raise MisconfigurationException(m)
 
+                # throw warning if key other than 'val_loss' is used in output
+                if not isinstance(output, EvalResult) or 'val_loss' not in output.keys():
```

Comment:
that’s not what we meant. 

the warning js when the word monitor changes in a callback AND you are using a result obj

Reply:
Oh I see. 
So something like when `EarlyStopping(monitor='not_val_loss')` AND using a `Result` object?

Reply:
correct